### DISCUSSION TITLE
Interpret OpenAPI examples and example fields

### DISCUSSION THREAD

**Discussion by adrien1212** [thunderclient/thunder-client-support#1076] [Q&A]
Does Thunder interpret `examples` and `example` fields ?
For example, in the specification of OpenAPI we can define examples for _body request_/_response_ 
SwaggerUI interpret it but I don't think Thunder do it.

Example :
```yaml
   post:
      operationId: create
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/DDClient'
            example:
              clientId: "un id"
              name: "un name"
```

So SwaggerUI interpret the example and add it automatically in the request 

![image](https://user-images.githubusercontent.com/24873705/227257472-783b25e2-c654-451a-937e-3ee14e189870.png)

But in Thunder there is only the schema without being completed with the example

![image](https://user-images.githubusercontent.com/24873705/227258372-e1fc4ec5-5094-44c2-b4ce-cf10a197c178.png)

So the user must fill it in by hand

**Comment by rangav** [comment]
Hi @adrien1212 can you share sample openAPI file, will verify it.

**Comment by adrien1212** [comment]
Hi @rangav 
Thank you, please find a Minimal reproducible example

```yaml
openapi: 3.0.0
info:
  description: Api to handle client in daily database
  title: Client api
  version: "1.0"
servers:
- url: /
tags:
- name: client
paths:
  /clients:
    post:
      operationId: create
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/ClientSearchCriteria'
            examples:
              default:
                $ref: '#/components/examples/defaultListRequest'
      responses:
        default:
          content:
            application/json:
              examples:
                default:
                  $ref: '#/components/examples/defaultListResponse'
              schema:
                items:
                  $ref: '#/components/schemas/ClientSearchCriteria'
                type: array
          description: default response

components:
  schemas:
    ClientSearchCriteria:
      properties:
        clientId:
          type:
            string
        name:
          type:
            string
      type: object
    
  examples:
    defaultListRequest:
      value:
        cliendId: "clientId body request"
        name: "name body request"
    defaultListResponse:
      value:
        cliendId: "clientId response"
        name: "name response"
```

### Thunder interpretation
![image](https://user-images.githubusercontent.com/24873705/227323601-d9cfc376-64e0-41b9-97f0-01cfc6b8111b.png)

It could be good to have instead
```yaml
{
  "clientId": "clientId body request",
  "name": "name body request"
}
```

**Reply by rangav** [reply]
Thanks for sharing will add to roadmap.

**Reply by rangav** [reply]
I have created a new feature request for this.
https://github.com/rangav/thunder-client-support/issues/1083
